content/en/blog/2021/better-external-authz/index.md

```diff
@@ -0,0 +1,545 @@
+external authorization on a proxy (either gateway or sidecar). The user should make sure the external auth service is up
+and running.
+
+In runtime,
+
+1. A request to the proxy will be paused, and the proxy will send a check request to the external auth service as
+   configured by the user in the authorization policy;
+
+1. The external auth service will check the request and make the decision whether to allow it or not;
+
+1. If allowed, the request will continue and still be enforced by the local authorization defined by ALLOW/DENY action.
+
+1. If denied, the request will be rejected immediately.
+
+The following is an example authorization policy with `CUSTOM` action:
+
+{{< text yaml >}}
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: ext-authz
+  namespace: istio-system
+spec:
+  # The selector applies to the ingress gateway in the istio-system namespace.
+  selector:
+    matchLabels:
+      app: istio-ingressgateway
+  # The action "CUSTOM" delegates the access control to an external authorizer, this is different from
+  # the ALLOW/DENY action that enforces the access control right inside the proxy.
+  action: CUSTOM
+  # The provider specifies the name of the external authorizer defined in the meshconfig, which tells where and how to
+  # talk to the external auth service. We will cover this more later.
+  provider:
+    name: "my-ext-authz-service"
+  # The rule specifies that the access control is triggered only if the request path has the prefix "/admin".
+  # This allows you to easily enable or disable the external authorization based on the requests, avoiding the external
+  # check request if it is not needed.
+  rules:
+  - to:
+    - operation:
+        paths: ["/admin/*"]
+{{< /text >}}
+
+The following is a snippet of the mesh config showing an example of the provider `my-ext-authz-service` definition:
+
+{{< text yaml >}}
+extensionProviders:
+# The name "my-ext-authz-service" is referred to by the authorization policy in its provider field.
+- name: "my-ext-authz-service"
+  # The "envoyExtAuthzGrpc" field specifies the type of the external authorization service is implemented by the Envoy
+  # ext-authz filter gRPC API. The other supported type is the Envoy ext-authz filter HTTP API.
+  # See more in https://www.envoyproxy.io/docs/envoy/v1.16.2/intro/arch_overview/security/ext_authz_filter.
+  envoyExtAuthzGrpc:
+    # The service and port specifies the address of the external auth service, "ext-authz.istio-system.svc.cluster.local"
+    # means the service is deployed in the mesh. It can also be defined out of the mesh or even inside the pod as a separate
+    # container.
+    service: "ext-authz.istio-system.svc.cluster.local"
+    port: 9000
+{{< /text >}}
+
+## Key Features
+
+The authorization policy of [`CUSTOM` action](https://github.com/istio/api/blob/b24ca4b76fdb51e227baf508c6d884eb8bcb51f5/security/v1beta1/authorization_policy.proto#L296)
```

Comment:
is this link correct when 1.9 is out?  (Same question on line 106 below)

Reply:
yes it points to the latest API that is shipped with 1.9.

Reply:
I meant, the URL is

`https://github.com/istio/api/blob/b24ca4b76fdb51e227baf508c6d884eb8bcb51f5/security/v1beta1/authorization_policy.proto#L296`

Should `b24ca4b76fdb51e227baf508c6d884eb8bcb51f5` be a tag or 'master' at the time of publication?

Reply:
I just updated to link to the istio.io reference page, so no need to include the tag or master at all.